lib/mix/lib/mix.ex

```diff
@@ -431,9 +431,9 @@ defmodule Mix do
   @doc """
   Raises a Mix error that is nicely formatted.
   """
-  @spec raise(binary) :: no_return
-  def raise(message) when is_binary(message) do
-    Kernel.raise(Mix.Error, mix: true, message: message)
+  @spec raise(binary, pos_integer) :: no_return
+  def raise(message, code \\ 1) when is_binary(message) do
+    Kernel.raise(Mix.Error, mix: true, message: message, code: code)
```

Comment:
I wonder if we should instead repurpose the `:mix` key. Something like this:

  def raise(message, opts \\ []) when is_binary(message) do
    Kernel.raise(Mix.Error, mix: Keyword.get(opts, :exit_code, 1), message: message)

And in the CLI, we check if mix is set. `true` is the same as 1.